src/main/java/domain/store/LottoStore.java

```diff
@@ -0,0 +1,32 @@
+package domain.store;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import domain.lotto.game.LottoGame;
+import domain.lotto.ticket.LottoGenerateStrategy;
+import domain.lotto.ticket.Lotto;
+
+public class LottoStore {
+	private static final long LOTTO_PRICE = 1_000L;
+	private LottoGenerateStrategy lottoGenerateStrategy;
+
+	public LottoStore(LottoGenerateStrategy lottoGenerateStrategy) {
+		this.lottoGenerateStrategy = lottoGenerateStrategy;
+	}
+
+	public LottoGame createGame(LottoPurchase lottoPurchase, List<Lotto> lottos) {
+		if (lottoPurchase.isNotEqualToManualAmount(lottos.size())) {
+			throw new IllegalArgumentException("수동 입력이 너무 많습니다.");
+		}
+		List<Lotto> mergelottos = new ArrayList<>(lottos);
+		for (int i = 0; i < getAutoAmount(lottoPurchase); i++) {
```

Comment:
```suggestion
		for (int i = 0, autoAmount = getAutoAmount(lottoPurchase); i < autoAmount; i++) {
```
`getAutoAmount` 메서드가 불필요하게 매번 호출될 것 같네요. effective java 3/e 아이템 57을 참고해보세요.

```java
for (int i = 0, n = expensiveComputation(); i < n; i++) {
  ... // i로 무언가를 한다.
}
```
> 다음은 지역 변수의 범위를 최소화하는 또 다른 반복문 관용구다.
> ...
> 반복 여부를 결정짓는 변수 i의 한곗값을 변수 n에 저장하여, 반복 때마다 다시 계산해야 하는 비용을 없앴다. 같은 값을 반환하는 메서드를 매번 호출한다면 이 관용구를 사용하기 바란다.

Reply:
배워갑니다.